tests/unit/processor/domain_resolver/test_domain_resolver.py

```diff
@@ -28,21 +27,159 @@ class TestDomainResolver(BaseProcessorTestCase):
         "tree_config": "tests/testdata/unit/shared_data/tree_config.json",
     }
 
-    def test_domain_to_ip_resolved_and_added(self, monkeypatch):
-        def mockreturn(domain):
-            if domain == "google.de":
-                return "1.2.3.4"
-            else:
-                return None
+    @mock.patch("socket.gethostbyname", return_value="1.2.3.4")
+    def test_domain_to_ip_resolved_and_added(self, _):
+        rule = {
+            "filter": "fqdn",
+            "domain_resolver": {"source_url_or_domain": "fqdn"},
+            "description": "",
+        }
+        self._load_specific_rule(rule)
+        document = {"fqdn": "google.de"}
+        expected = {"fqdn": "google.de", "resolved_ip": "1.2.3.4"}
+        self.object.process(document)
+        assert document == expected
+
+    @mock.patch("socket.gethostbyname", return_value="1.2.3.4")
+    def test_domain_to_ip_resolved_and_added_from_cache(self, _):
+        rule = {
+            "filter": "fqdn",
+            "domain_resolver": {"source_url_or_domain": "fqdn"},
+            "description": "",
+        }
+        self._load_specific_rule(rule)
+        document = {"fqdn": "google.de"}
+        expected = {"fqdn": "google.de", "resolved_ip": "1.2.3.4"}
+        self.object.process(document)
+        assert document == expected
+
+    @mock.patch("socket.gethostbyname", return_value="1.2.3.4")
+    def test_url_to_ip_resolved_and_added(self, _):
+        rule = {
+            "filter": "url",
+            "domain_resolver": {"source_url_or_domain": "url"},
+            "description": "",
+        }
+        self._load_specific_rule(rule)
+        document = {"url": "https://www.google.de/something"}
+        expected = {"url": "https://www.google.de/something", "resolved_ip": "1.2.3.4"}
+        self.object.process(document)
+        assert document == expected
 
-        monkeypatch.setattr(socket, "gethostbyname", mockreturn)
+    def test_domain_ip_map_greater_cache(self):
```

Comment:
This test doesn't have any assertion anymore. What is it testing for or what behavior is expected?

Reply:
done